server/models/service-process.js

```diff
@@ -0,0 +1,132 @@
+var async = require('async');
+var debug = require('debug')('strong-mesh-models:server:service-process');
+
+module.exports = function(ServiceProcess) {
+  function recordFork(instanceId, pInfo, callback) {
+    debug('Process forked: worker id %d pid %d ppid %d',
+      pInfo.id, pInfo.pid, pInfo.ppid);
+
+    var proc = new ServiceProcess({
+      pid: pInfo.pid,
+      parentPid: pInfo.ppid,
+      workerId: pInfo.id,
+      serviceInstanceId: instanceId,
+      startTime: new Date()
+    });
+
+    proc.save(function(err, proc) {
+      debug('upsert Process: %j', err || proc);
+      if (err) return callback(err);
+      callback();
+    });
+  }
+  ServiceProcess.recordFork = recordFork;
+
+  function recordExit(instanceId, pInfo, callback) {
+    debug('Process exited: worker id %d pid %d reason %s suicide? %s',
+      pInfo.id, pInfo.pid, pInfo.reason, pInfo.suicide);
+
+    async.waterfall([
```

Comment:
More of an "FYI" than a "please change"; one of the conventions I follow is it to explicitly `return` on lines like these so the person reading the code knows that this is the intended end of the function.